utilities-core/src/test/java/uk/gov/justice/services/yaml/YamlParserTest.java

```diff
@@ -0,0 +1,169 @@
+        };
+        final Map<String, SubscriptionDescriptor> subscriptionDescriptorDefinitionMap = yamlParser().parseYamlFrom(url, typeReference);
+        final SubscriptionDescriptor subscriptionsDescriptor = subscriptionDescriptorDefinitionMap.get("subscriptions_descriptor");
+        assertThat(subscriptionsDescriptor.getService(), is("examplecontext"));
+        assertThat(subscriptionsDescriptor.getServiceComponent(), is("EVENT_LISTENER"));
+        assertThat(subscriptionsDescriptor.getSpecVersion(), is("1.0.0"));
+
+        final List<Subscription> subscriptions = subscriptionsDescriptor.getSubscriptions();
+        assertThat(subscriptions.size(), is(2));
+
+        final Subscription subscription_1 = subscriptions.get(0);
+        assertThat(subscription_1.getName(), is("subscription1"));
+        assertThat(subscription_1.getEventSourceName(), is("example"));
+
+        final List<Event> events_1 = subscription_1.getEvents();
+        assertThat(events_1.size(), is(2));
+        assertThat(events_1.get(0).getName(), is("example.recipe-added"));
+        assertThat(events_1.get(0).getSchemaUri(), is("http://justice.gov.uk/json/schemas/domains/example/example.recipe-added.json"));
+        assertThat(events_1.get(1).getName(), is("example.recipe-deleted"));
+        assertThat(events_1.get(1).getSchemaUri(), is("http://justice.gov.uk/json/schemas/domains/example/example.recipe-deleted.json"));
+    }
+
+
+    @Test
+    public void shouldThrowFileNotFoundExceptionForTypeReference() throws Exception {
+
+        final URL url = get("this-subscription-does-not-exist.yaml").toUri().toURL();
+        final TypeReference<SubscriptionDescriptor> typeReference
+                = new TypeReference<SubscriptionDescriptor>() {
+        };
+
+        try {
+            yamlParser().parseYamlFrom(url, typeReference);
+            fail();
+        } catch (final YamlParserException e) {
+            assertThat(e.getCause(), is(instanceOf(FileNotFoundException.class)));
+            assertThat(e.getMessage(), containsString("Failed to read YAML file"));
+            assertThat(e.getMessage(), containsString("this-subscription-does-not-exist.yaml"));
+        }
+    }
+
+    @Test
+    public void shouldThrowFileNotFoundExceptionForClassType() throws Exception {
+
+        final URL url = get("this-subscription-does-not-exist.yaml").toUri().toURL();
+        try {
+            yamlParser().parseYamlFrom(url, SubscriptionDescriptor.class);
+            fail();
+        } catch (final YamlParserException e) {
+            assertThat(e.getCause(), is(instanceOf(FileNotFoundException.class)));
+            assertThat(e.getMessage(), containsString("Failed to read YAML file"));
+            assertThat(e.getMessage(), containsString("this-subscription-does-not-exist.yaml"));
+        }
+    }
+
+    private YamlParser yamlParser() {
+        final YamlParser yamlParser = new YamlParser();
+        yamlParser.objectMapperFactory = new ObjectMapperFactory();
```

Comment:
If the ObjectMapperFactory needs injecting use @Spy with mocking, or ReflectionUtils with a new instance.  That way the injection can be private.